src/vlib/trace.c

```diff
@@ -265,6 +270,36 @@ trace_apply_filter (vlib_main_t * vm)
   vec_free (traces_to_remove);
 }
 
+/*
+ * Filters packets based on specific characteristics.
+ * NB: Not to be confused with "trace filter"
+ */
+u32
+show_trace_pkt_filter (vlib_show_trace_filter_t *filters, vlib_trace_header_t *h)
+{
+  if (vec_len(filters) == 0)
+    return 1;
+  
+  /* Iterate through the nodes to see if we have a match on the filters */
+  vlib_trace_header_t *end = vec_end(h);
+  vlib_show_trace_filter_t *filter;
+  u8 matches = 0;
+  
+  while (h < end)
+    {
+      vec_foreach(filter, filters) {
+	if (h->node_index != filter->node_index)
```

Comment:
This is just a question so that I can understand this code better (and not review feedback): does the `node_index` test here ensure that the `memcmp` a few lines down compare the same data? (that is, the same type of header)

And the condition at the end comparing `matches` is ok because any given filter can only apply to at most one header in the trace header chain right?

Reply:
Correct.

The filter will have the data and the offsets for a specific node e.g. ip4-input (in the case of ip4 src, dst and port). Each filter can only be matched once, thus if matches equal the number of filters it means all the filters have been matched.